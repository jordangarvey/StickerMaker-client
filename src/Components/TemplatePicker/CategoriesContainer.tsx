import React, { FC } from "react";
import { ScrollView, StyleSheet, View } from "react-native";

import Categories from "./Categories";

import Category from "./Category";


/**
 * Interface for the Categories Container props
 */
interface ICategoriesContainerProps {
	/** Callback to be fired whenever the option button is pressed */
	onPress(category: Categories): void;

	/** The currently selected category */
	currentCategory?: Categories;
}

const CategoriesContainer: FC<ICategoriesContainerProps> = (props) => (
	<View style={styles.categoriesContainer}>
		<ScrollView horizontal={true}>
			{
				Object.keys(Categories).map((category: any) => (
					<Category
						key={Categories[category]}
						category={Categories[category] as Categories}
						onPress={props.onPress}
						selected={Categories[category] === props.currentCategory}
					/>
				))
			}
		</ScrollView>
	</View>
);

/**
 * Styles for the Categories Container
 */
const styles = StyleSheet.create({
	categoriesContainer: {
		marginBottom: 20,
		width: "100%"
	}
});

export default CategoriesContainer;
