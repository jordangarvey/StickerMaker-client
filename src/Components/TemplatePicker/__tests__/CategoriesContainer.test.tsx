import React from "react";

import renderer from "react-test-renderer";

import CategoriesContainer from "../CategoriesContainer";


describe("The Categories Container componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<CategoriesContainer
				onPress={mockFunction}
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
