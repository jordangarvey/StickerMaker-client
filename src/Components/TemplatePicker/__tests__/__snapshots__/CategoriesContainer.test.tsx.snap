// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Categories Container componentâ€™s snapshot renders as expected 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "marginHorizontal": 20,
            "opacity": 1,
            "padding": 5,
            "width": 60,
          }
        }
      >
        <Text
          style={false}
        >
          Colour
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "marginHorizontal": 20,
            "opacity": 1,
            "padding": 5,
            "width": 60,
          }
        }
      >
        <Text
          style={false}
        >
          Shape
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "marginHorizontal": 20,
            "opacity": 1,
            "padding": 5,
            "width": 60,
          }
        }
      >
        <Text
          style={false}
        >
          Text
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
